Update 'Configure firmware' step

MLopezJ opened the pull request:
Right now, there are some confusions about where some steps should occur. 

#478 implement some changes to make visible the working directory where the commands should be executed and this PR add fixes to detail where to locate files and specify clarification to avoid circular confusion between `Configure the firmware` step and `build firmware` options.

**Circular confusion:**
If you are following the documentation (https://nordicsemiconductor.github.io/asset-tracker-cloud-docs/saga/docs/aws/GettingStarted/Index.html), the `Configure the firmware` step has an issue, because at some point you will need to compile the firmware and to do that you have some options, however that will take you to another page and if after finish that process you click `next`, it will show you other building options but not to the `Create device credentials`.

This is the solution I'am proposing, however I'm open to discuss it and find new ways.

Lenakh97 commented on docs/aws/GettingStarted/FirmwareConfiguration.rst:
```diff
@@ -5,12 +5,13 @@ Configure the firmware
 To configure the :ref:`firmware <firmware-aws-index>`, complete the following mandatory steps:
 
-1. Create the file :file:`firmware.conf` to be used for your configuration overrides.
+1. Create the file :file:`firmware.conf` into ``~/nrf-asset-tracker`` folder to be used in the future for your configuration overrides.
```
```suggestion
1. Create the file :file:`firmware.conf` into the ``~/nrf-asset-tracker`` folder to be used in the future for your configuration overrides.
```

peknis replied:
```suggestion
1. Create the file :file:`firmware.conf` into the :file:`~/nrf-asset-tracker` folder for later use as your configuration overrides.
```

peknis commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,7 @@ To clone the firmware repository, run the following command:
       https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-aws firmware
 
-Then, follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+Then, move the :file:`firmware.conf` file created on the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` step into ``~/nrf-asset-tracker/firmware``.
```
```suggestion
Then, move the :file:`firmware.conf` file created in the :ref:`firmware configuration phase <aws-firmware-configuration>` into `:file:`~/nrf-asset-tracker/firmware`.
```
Do the same changes to the other docs as well.

peknis replied:
Right, Uma's proposals were way better than mine.

peknis commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -52,3 +52,8 @@ Location of the HEX file
+Device credentials
+=====================
+
+Click :ref:`Create device credentials <aws-device-credentials>` to continue with the configuration or click ``Next`` to see other ``firmware`` build options. 
```
Hold on, where are we now? What are we clicking and where? If we refer to a doc page, we don't say click. Not that sure we need this section here or any of the docs in this PR.

coderbyheart replied:
I reviewed the changes, and the point is to bring users back to the Getting Started section, because the navigate outside of it for building the firmware.

MLopezJ committed:
fix: update 'configure the firmware' step

Add details about where to locate files and add link for users
navigating from Getting Started section to firmware building section
back to Getting Started guide.

Co-authored-by: Markus Tacker <markus.tacker@nordicsemi.no>

MLopezJ committed:
fix: update 'configure the firmware' step

specify working directory where a command should be executed

wiba-nordic commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -52,3 +60,8 @@ Location of the HEX file
+Device credentials
+******************
+
+For the device to be able to connect to the nRF Asset Tracker for AWS, now :ref:`create device credentials <aws-device-credentials>`.
```
```suggestion
For the device to be able to connect to the nRF Asset Tracker for AWS, you must :ref:`create device credentials <aws-device-credentials>`.
```

MLopezJ committed:
fix: update text description

Update docs/firmware/aws/BuildingUsingDocker.rst

Co-authored-by: Wille Backman <wille.backman@nordicsemi.no>

umapraseeda commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,15 @@ To clone the firmware repository, run the following command:
       https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-aws firmware
 
-Then, follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+If you haven't configured the firmware yet, follow the :ref:`configure the firmware instructions <aws-firmware-configuration>`.
```
```suggestion
If you have not configured the firmware yet, follow the :ref:` instructions for configuring the firmware <aws-firmware-configuration>`.
```

umapraseeda commented on docs/firmware/aws/BuildingUsingLocalSystem.rst:
```diff
@@ -41,7 +42,15 @@ Create a folder, for example, ``ncs`` and initialize the project by running the
 *********************
 
-Follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file.
+If you haven't configured the firmware yet, follow the :ref:`configure the firmware instructions <aws-firmware-configuration>`.
```
```suggestion
If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <aws-firmware-configuration>`.
```

umapraseeda commented on docs/firmware/azure/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,15 @@ To clone the firmware repository, run the following command:
       https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-azure firmware
 
-Then, follow the :ref:`configuration instructions for the firmware <azure-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+If you haven't configured the firmware yet, follow the :ref:`configure the firmware instructions <azure-firmware-configuration>`.
```
```suggestion
If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <azure-firmware-configuration>`.
```

umapraseeda commented on docs/firmware/azure/BuildingUsingLocalSystem.rst:
```diff
@@ -40,7 +42,15 @@ Create a folder, for example, ``ncs`` and initialize the project by running the
 *********************
 
-Follow the :ref:`configuration instructions for the firmware <azure-firmware-configuration>` and add your settings to the :file:`firmware.conf` file.
+If you haven't configured the firmware yet, follow the :ref:`configure the firmware instructions <azure-firmware-configuration>`.
```
```suggestion
If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <azure-firmware-configuration>`.
```

MLopezJ committed:
fix: update text description

Update docs/firmware/aws/BuildingUsingDocker.rst

Co-authored-by: Uma Praseeda <55196231+umapraseeda@users.noreply.github.com>

MLopezJ committed:
fix: update text description

Update docs/firmware/aws/BuildingUsingLocalSystem.rst

Co-authored-by: Uma Praseeda <55196231+umapraseeda@users.noreply.github.com>

MLopezJ committed:
fix: update text description

Update docs/firmware/azure/BuildingUsingDocker.rst

Co-authored-by: Uma Praseeda <55196231+umapraseeda@users.noreply.github.com>

MLopezJ committed:
fix: update text description

Update docs/firmware/azure/BuildingUsingLocalSystem.rst

Co-authored-by: Uma Praseeda <55196231+umapraseeda@users.noreply.github.com>

umapraseeda commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,15 @@ To clone the firmware repository, run the following command:
       https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-aws firmware
 
-Then, follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+If you have not configured the firmware yet, follow the :ref:` instructions for configuring the firmware <aws-firmware-configuration>`.
```
An extra space
```suggestion
If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <aws-firmware-configuration>`.
```

peknis commented on docs/aws/GettingStarted/FirmwareConfiguration.rst:
```diff
@@ -5,12 +5,14 @@ Configure the firmware
 To configure the :ref:`firmware <firmware-aws-index>`, complete the following mandatory steps:
 
-1. Create the file :file:`firmware.conf` to be used for your configuration overrides.
+1. Create the file :file:`firmware.conf` in the ``~/nrf-asset-tracker`` folder for your configuration overrides.
```
```suggestion
1. Create the file :file:`firmware.conf` in the :file:`~/nrf-asset-tracker` folder for your configuration overrides.
```

peknis commented on docs/azure/GettingStarted/FirmwareConfiguration.rst:
```diff
@@ -5,12 +5,14 @@ Configure the firmware
 To configure the :ref:`firmware <firmware-azure-index>`, complete the following mandatory steps:
 
-1. Create the file :file:`firmware.conf` to be used for your configuration overrides.
+1. Create the file :file:`firmware.conf` in the ``~/nrf-asset-tracker`` folder for your configuration overrides.
```
```suggestion
1. Create the file :file:`firmware.conf` in the :file:`~/nrf-asset-tracker` folder for your configuration overrides.
```

peknis commented on docs/firmware/aws/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,15 @@ To clone the firmware repository, run the following command:
-Then, follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+If you have not configured the firmware yet, follow the :ref:` instructions for configuring the firmware <aws-firmware-configuration>`.
+
+Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into ``~/nrf-asset-tracker/firmware``.
```
```suggestion
Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into the :file:`~/nrf-asset-tracker/firmware` folder.
```

peknis commented on docs/firmware/aws/BuildingUsingLocalSystem.rst:
```diff
@@ -41,7 +42,15 @@ Create a folder, for example, ``ncs`` and initialize the project by running the
-Follow the :ref:`configuration instructions for the firmware <aws-firmware-configuration>` and add your settings to the :file:`firmware.conf` file.
+If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <aws-firmware-configuration>`.
+
+Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into ``~/nrf-asset-tracker/ncs/firmware``.
```
```suggestion
Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into the :file:`~/nrf-asset-tracker/ncs/firmware` folder.
```

peknis commented on docs/firmware/azure/BuildingUsingDocker.rst:
```diff
@@ -25,7 +25,15 @@ To clone the firmware repository, run the following command:
-Then, follow the :ref:`configuration instructions for the firmware <azure-firmware-configuration>` and add your settings to the :file:`firmware.conf` file in the ``firmware`` folder.
+If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <azure-firmware-configuration>`.
+
+Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into ``~/nrf-asset-tracker/firmware``.
```
```suggestion
Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into the :file:`~/nrf-asset-tracker/firmware` folder.
```

peknis commented on docs/firmware/azure/BuildingUsingLocalSystem.rst:
```diff
@@ -40,7 +42,15 @@ Create a folder, for example, ``ncs`` and initialize the project by running the
-Follow the :ref:`configuration instructions for the firmware <azure-firmware-configuration>` and add your settings to the :file:`firmware.conf` file.
+If you have not configured the firmware yet, follow the :ref:`instructions for configuring the firmware <azure-firmware-configuration>`.
+
+Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into ``~/nrf-asset-tracker/ncs/firmware``.
```
```suggestion
Then, move the :file:`~/nrf-asset-tracker/firmware.conf` file into the :file:`~/nrf-asset-tracker/ncs/firmware` folder.
```

MLopezJ committed:
fix: add extra space in text

Update docs/firmware/aws/BuildingUsingDocker.rst

Co-authored-by: Uma Praseeda <55196231+umapraseeda@users.noreply.github.com>

MLopezJ committed:
fix: update semantic specification

Update docs/aws/GettingStarted/FirmwareConfiguration.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

MLopezJ committed:
fix: update semantic specification

Update docs/azure/GettingStarted/FirmwareConfiguration.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

MLopezJ committed:
fix: update semantic and text specification

Update docs/firmware/aws/BuildingUsingDocker.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

MLopezJ committed:
fix: update semantic and text specification

Update docs/firmware/aws/BuildingUsingLocalSystem.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

MLopezJ committed:
fix: update semantic and text specification

Update docs/firmware/azure/BuildingUsingDocker.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

MLopezJ committed:
fix: update semantic and text specification

Update docs/firmware/azure/BuildingUsingLocalSystem.rst

Co-authored-by: Pekka Niskanen <73991005+peknis@users.noreply.github.com>

wiba-nordic commented on docs/firmware/azure/BuildingUsingLocalSystem.rst:
```diff
@@ -65,3 +75,8 @@ Location of the HEX file
+Device credentials
+******************
+
+For the device to be able to connect to the nRF Asset Tracker for Azure, now :ref:`create device credentials <azure-device-credentials>`.
```
```suggestion
For the device to be able to connect to the nRF Asset Tracker for Azure, you must :ref:`create device credentials <azure-device-credentials>`.
```

wiba-nordic commented on docs/firmware/azure/BuildingUsingDocker.rst:
```diff
@@ -52,3 +60,8 @@ Location of the HEX file
+Device credentials
+******************
+
+For the device to be able to connect to the nRF Asset Tracker for Azure, now :ref:`create device credentials <azure-device-credentials>`.
```
```suggestion
For the device to be able to connect to the nRF Asset Tracker for Azure, you must :ref:`create device credentials <azure-device-credentials>`.
```

wiba-nordic commented on docs/firmware/aws/BuildingUsingLocalSystem.rst:
```diff
@@ -66,3 +75,8 @@ Location of the HEX file
+Device credentials
+******************
+
+For the device to be able to connect to the nRF Asset Tracker for AWS, now :ref:`create device credentials <aws-device-credentials>`.
```
```suggestion
For the device to be able to connect to the nRF Asset Tracker for AWS, you must :ref:`create device credentials <aws-device-credentials>`.
```

MLopezJ committed:
fix: update text description

Be specific on required step. Update docs/firmware/azure/BuildingUsingLocalSystem.rst

Co-authored-by: Wille Backman <wille.backman@nordicsemi.no>

MLopezJ committed:
fix: update text description

Be specific on required step. Update docs/firmware/azure/BuildingUsingDocker.rst

Co-authored-by: Wille Backman <wille.backman@nordicsemi.no>

MLopezJ committed:
fix: update text description

Be specific on required step. Update docs/firmware/aws/BuildingUsingLocalSystem.rst

Co-authored-by: Wille Backman <wille.backman@nordicsemi.no>
